packages/netdog/netdog.spec

```diff
@@ -0,0 +1,133 @@
+Requires: %{_cross_os}systemd-resolved
+Requires: %{_cross_os}netdog
+Conflicts: (%{_cross_os}netdog-wicked or %{_cross_os}image-feature(no-systemd-networkd))
+%description -n %{_cross_os}netdog-systemd-networkd
+%{summary}.
+
+%package -n %{_cross_os}netdog-wicked
+Summary: Bottlerocket network configuration helper
+Provides: %{_cross_os}netdog = 1
+Requires: %{_cross_os}netdog-common
+Requires: %{_cross_os}wicked
+Requires: %{_cross_os}netdog
+Conflicts: (%{_cross_os}netdog-systemd-networkd or %{_cross_os}image-feature(systemd-networkd))
+%description -n %{_cross_os}netdog-wicked
+%{summary}.
+
+%package -n %{_cross_os}hostname-reverse-dns
+Summary: Reverse DNS Hostname detector
+%description -n %{_cross_os}hostname-reverse-dns
+%{summary}
+
+%package -n %{_cross_os}hostname-imds
+Summary: IMDS Hostname detector
+%description -n %{_cross_os}hostname-imds
+%{summary}
+
+%prep
+%setup -T -c
+%cargo_prep
+
+%build
+mkdir bin
+
+echo "** Build Dogtag Hostname Helpers"
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p dogtag \
+    --bins \
+    --target-dir=${HOME}/.cache/dogtag
+
+echo "** Build Netdog Binaries"
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p netdog \
+    --features default \
+    --target-dir=${HOME}/.cache/networkd
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p netdog \
+    --features wicked \
+    --target-dir=${HOME}/.cache/wicked
+
+%install
+install -d %{buildroot}%{_cross_libexecdir}/hostname-resolvers
+install -p -m 0755 ${HOME}/.cache/dogtag/%{__cargo_target}/release/20-imds %{buildroot}%{_cross_libexecdir}/hostname-resolvers/20-imds
+install -p -m 0755 ${HOME}/.cache/dogtag/%{__cargo_target}/release/10-reverse-dns %{buildroot}%{_cross_libexecdir}/hostname-resolvers/10-reverse-dns
+
+install -d %{buildroot}%{_cross_bindir}
+install -p -m 0755 ${HOME}/.cache/networkd/%{__cargo_target}/release/netdog %{buildroot}%{_cross_bindir}/netdog-systemd-networkd
+install -p -m 0755 ${HOME}/.cache/wicked/%{__cargo_target}/release/netdog %{buildroot}%{_cross_bindir}/netdog-wicked
+
+install -d %{buildroot}%{_cross_tmpfilesdir}
+install -p -m 0644 %{S:0} %{buildroot}%{_cross_tmpfilesdir}/netdog.conf
+
+install -d %{buildroot}%{_cross_unitdir}
+install -p -m 0644 %{S:10} %{S:11} %{S:12} %{S:13} %{buildroot}%{_cross_unitdir}
+
+install -d %{buildroot}%{_cross_libdir}
+install -d %{buildroot}%{_cross_libdir}/systemd/resolved.conf.d
+install -p -m 0644 %{S:20} %{buildroot}%{_cross_libdir}/systemd/resolved.conf.d
+
+%post -n %{_cross_os}netdog-wicked -p <lua>
+posix.symlink("netdog-wicked", "%{_cross_bindir}/netdog")
+
+%post -n %{_cross_os}netdog-systemd-networkd -p <lua>
+posix.symlink("netdog-systemd-networkd", "%{_cross_bindir}/netdog")
+
+%files -n %{_cross_os}hostname-reverse-dns
+%{_cross_libexecdir}/hostname-resolvers/10-reverse-dns
+
+%files -n %{_cross_os}hostname-imds
+%{_cross_libexecdir}/hostname-resolvers/20-imds
+
+%files -n %{_cross_os}netdog-common
```

Comment:
Actually, you don't have anything in the `netdog` package now, and we don't need to play the weird game with epochs, so I would just drop the "common" subpackage and move this stuff into `%files` (for the main netdog package).